shell/platform/android/apk_asset_provider.cc

```diff
@@ -0,0 +1,38 @@
+#include <unistd.h>
+#include <algorithm>
+#include <sstream>
+
+#include "flutter/shell/platform/android/apk_asset_provider.h"
+#include "lib/fxl/logging.h"
+
+namespace blink {
+
+bool APKAssetProvider::GetAsBuffer(const std::string& asset_name,
+                                   std::vector<uint8_t>* data) {
+  std::stringstream ss;
+  ss << directory_.c_str() << "/" << asset_name;
+  AAsset* asset = AAssetManager_open(assetManager_, ss.str().c_str(), AASSET_MODE_RANDOM);
```

Comment:
I don't think we need random access to the asset bytes. We only want to fill a buffer, right?
So I'd expect this to be AASSET_MODE_BUFFER (though I haven't tried it).

https://developer.android.com/ndk/reference/group___asset.html